app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -1,4 +1,64 @@
-class ManageIQ::Providers::Vmware::InfraManager::OrchestrationTemplate < OrchestrationTemplate
+class ManageIQ::Providers::Vmware::InfraManager::OrchestrationTemplate < ::OrchestrationTemplate
+  belongs_to :ext_management_system, :foreign_key => "ems_id", :class_name => "ManageIQ::Providers::Vmware::InfraManager"
+
+  SPEC_KEY_MAPPING = {
+    "resource_pool" => "resource_pool_id",
+    "ems_folder"    => "folder_id",
+    "host"          => "host_id"
+  }.freeze
+
+  def connect
+    require 'vsphere-automation-cis'
+
+    configuration = VSphereAutomation::Configuration.new.tap do |c|
+      c.host = ext_management_system.hostname
+      c.username = ext_management_system.auth_user_pwd.first
+      c.password = ext_management_system.auth_user_pwd.last
+      c.verify_ssl = false
+      c.verify_ssl_host = false
+    end
+
+    api_client = VSphereAutomation::ApiClient.new(configuration)
+    VSphereAutomation::CIS::SessionApi.new(api_client).create('')
+    api_client
+  end
+
+  def with_provider_connection
+    raise _("no block given") unless block_given?
+    _log.info("Connecting through #{ext_management_system.class.name}: [#{ext_management_system.name}]")
+    yield connect
+  end
+
+  def deploy(options = {})
+    require 'vsphere-automation-vcenter'
```

Comment:
not sure this require is needed here, since the code below calls `with_provider_connection`, which calls `connect`, which does the same `require`